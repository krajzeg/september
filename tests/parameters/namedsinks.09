inventory := |:::counts| {
	print("Apples:", counts.apples)
	print("Bananas:", counts.bananas)
	print("Cucumbers:", counts.cucumbers)
}

print("Our inventory should have 1 apple, 2 bananas and 3 cucumbers.")
inventory(apples: 1, bananas: 2, cucumbers: 3)

print("Let's try again by passing the argument explicitly.")
counts := Object()
counts::apples = 1
counts::bananas = 2
counts::cucumbers = 3
inventory(counts: counts)

print("But we shouldn't be able to pass a parameter both explicitly and implicitly.")
try {
	inventory(counts: counts, apples: 42)
	print("But we are.")
} catch (EWrongArguments) {
	print("And we aren't.")
}

print("Or pass the same argument twice.")
try {
	inventory(apples: 2, apples:3, bananas:4, cucumbers: 5)
	print("But we are.")
} catch (EWrongArguments) {
	print("And we aren't.")
}

mixed_inventory := |dactyls, :::others| {
	print("Dactyls: ", dactyls)
	print("Eggplants: ", others.eggplants)
}
print("Mixing named sinks with positional arguments should work.")
mixed_inventory(eggplants: 5, 4)
print("Mixing named sinks with other named arguments should also work.")
mixed_inventory(eggplants: 5, dactyls: 4)


