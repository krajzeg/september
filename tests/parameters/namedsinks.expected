Our inventory should have 1 apple, 2 bananas and 3 cucumbers.
Apples: 1
Bananas: 2
Cucumbers: 3
Let's try again by passing the argument explicitly.
Apples: 1
Bananas: 2
Cucumbers: 3
But we shouldn't be able to pass a parameter both explicitly and implicitly.
And we aren't.
Or pass the same argument twice.
And we aren't.
Mixing named sinks with positional arguments should work.
Dactyls:  4
Eggplants:  5
Mixing named sinks with other named arguments should also work.
Dactyls:  4
Eggplants:  5
